compiler/rustc_mir_dataflow/src/impls/liveness.rs

```diff
@@ -179,6 +179,9 @@ impl DefUse {
                 }
             }
 
+            // This terminator writes a new value, so it's a Def
+            PlaceContext::MutatingUse(MutatingUseContext::DropAndReplace) => Some(DefUse::Def),
```

Comment:
I would expect to find this case together with arm above, i.e., it is a def but only when it is a direct place without projections. Anyway, is this actually reachable?

Reply:
Ah, my suggestion wasn't quite right either. The drop precedes the assignment in the evaluation order, so even if there were a definition, the overall effect would be a use of a local regardless.